src/target/samd.c

```diff
@@ -526,9 +545,15 @@ bool samd_probe(target_s *t)
 	/* Protected? */
 	const bool protected = (ctrlstat & SAMD_STATUSB_PROT);
 
-	snprintf(priv_storage->samd_variant_string, SAMD_VARIANT_STR_LENGTH, "Atmel SAM%c%02d%c%d%c%s (rev %c)%s",
-		samd.family, samd.series, samd.pin, samd.mem, samd.variant, samd.package, samd.revision,
-		protected ? " protected" : "");
+	if (samd.series == 7) {
+		snprintf(priv_storage->samd_variant_string, SAMD_VARIANT_STR_LENGTH,
+			"Microchip PIC32CM%02d%02dMC000%02d (rev %c)%s", (samd.mem > 8) ? 12 : 64, samd.mem, samd.pin,
```

Comment:
The parens in the ternary are superfluous and should be dropped. They inhibit a diagnostic we want for accidental assignment. `samd.mem > 8U ? 12U : 64U` is perfectly valid, and the operators all bind how you'd hope.

Please adjust the format string to use `%u` not `%d` for unsigned values, as `%d` is strictly for signed numbers. The compiler should be complaining about this and if it is not we should adjust the compilation options to make it (job for us to do, not on you).